### DISCUSSION TITLE
Duplicity backups with mega

### DISCUSSION THREAD

**Discussion by rquevedo** [Tecnativa/doodba#383] [General]
Im trying to use Mega as Duplicity backend. I have two questions:
1. How can i pass user and password to duplicity url. Currently only username is used because other @ in ulr ends with error:
mega://somemegauser@mega.nz/BACKUPS
2. When im try to run backup i get this error:
+ exec dup /mnt/backup/src mega://somemegauser@mega.nz/BACKUPS
BackendException: command 'megals' not found, make sure megatools are installed

In witch container i have to install megatools?

**Comment by joao-p-marques** [comment]
Hi @rquevedo 

We use https://github.com/Tecnativa/docker-duplicity as the image for the automated backups container. That bundles duplicity and some common backends for backups (like S3). If you need to add a backend to be supported by duplicity, you need to add that to the image. You can create your own image from ours (`FROM ghcr.io/tecnativa/docker-duplicity`) and install the needed package or, if you think the community might benefit from that addition, you can also open a PR to https://github.com/Tecnativa/docker-duplicity to add that in the base image. You can check https://github.com/Tecnativa/docker-duplicity#managing-packages for more info.

**Reply by rquevedo** [reply]
Hi @joao-p-marques, 
Once again thank you for take time to answer my questions. Taking you advice i have compiled megatools in my new image. I had to change commons.yaml and add:
build: './backup'
Then in my project mkdir a backup folder with Dockerfile inside:

FROM tecnativa/duplicity:postgres

RUN apk add --update build-base libcurl curl-dev asciidoc openssl-dev glib-dev glib libtool automake autoconf
RUN rm -rf /src
RUN mkdir -p /src
WORKDIR /src
RUN wget https://megatools.megous.com/builds/megatools-1.10.2.tar.gz
RUN tar -xzvf megatools-1.10.2.tar.gz megatools-1.10.2
WORKDIR /src/megatools-1.10.2
RUN ./configure --prefix=$HOME/.local
RUN make -j4
RUN make install
RUN rm -rf /usr/local/bin/megacopy
RUN ln -s /src/megatools-1.10.2/megacopy /usr/local/bin/megacopy
RUN rm -rf /usr/local/bin/megadf
RUN ln -s /src/megatools-1.10.2/megadf /usr/local/bin/megadf
RUN rm -rf /usr/local/bin/megadl
RUN ln -s /src/megatools-1.10.2/megadl /usr/local/bin/megadl
RUN rm -rf /usr/local/bin/megaget
RUN ln -s /src/megatools-1.10.2/megaget /usr/local/bin/megaget
RUN rm -rf /usr/local/bin/megals
RUN ln -s /src/megatools-1.10.2/megals /usr/local/bin/megals
RUN rm -rf /usr/local/bin/megamkdir
RUN ln -s /src/megatools-1.10.2/megamkdir /usr/local/bin/megamkdir
RUN rm -rf /usr/local/bin/megaput
RUN ln -s /src/megatools-1.10.2/megaput /usr/local/bin/megaput
RUN rm -rf /usr/local/bin/megareg
RUN ln -s /src/megatools-1.10.2/megareg /usr/local/bin/megareg
RUN rm -rf /usr/local/bin/megarm
RUN ln -s /src/megatools-1.10.2/megarm /usr/local/bin/megarm
WORKDIR /
RUN touch /root/.megarc
RUN echo "[Login]" >> /root/.megarc
RUN echo "Username = someemail@gmail.com" >> /root/.megarc
RUN echo "Password = somepassword" >> /root/.megarc

I dont know if this is the better way to accomplish that but this is the way i found over many post and forums.
Any advice will be appreciated.

**Reply by joao-p-marques** [reply]
@rquevedo seems like a good option if it works for you. A few improvements could be made to the Dockerfile syntax but that is not mandatory.
For example, you probably don't need to remove each of the `/usr/local/bin/mega*` binaries, as they will not exist in the base image, and I am not sure if you actually need to link all the binaries to `/usr/local/bin`, as it will be a library used by duplicity, not a user binary. You could also configure the Username and Password through Docker environment variables, instead of hardcoding it on the Dockerfile. 
Overall, you could test this options, but if the current implementation works for you, great!

**Reply by rquevedo** [reply]
Hi @joao-p-marques i will try all your advices, all are true. About username and password in .megarc i have to copy manually after container is up because megatools search for .megarc in /root and /root is declared as volume in this image and all stuff write in Dockerfile to this folder are ignored.
Thanks again.

**Reply by pleirb** [reply]
Hello everyone! I'm starting with this doodba thing, I've been experimenting for some time and I'm at the stage of wanting to put it into production, without first making sure that the backups work. I want to use the recommended methods and Duplicity seems to offer several options, the doodba-copier-template only comes ready for Amazon S3, is it so? What if you have Google Drive, Mega or PCloud?
On the other hand, the image that doodba uses is docker-duplicity-postgres and I understand that it is to support the postgreSQL database, but what about the odoo filestore? Maybe someone can clear up my confusion. Thank you !
